pkg/providers/aws/cloudwatch.go

```diff
@@ -15,52 +15,58 @@ import (
 	v1 "github.com/re-cinq/aether/pkg/types/v1"
 )
 
-// Get the resource consumption of an ec2 instance
+const cpuExpression = `SELECT AVG(CPUUtilization) FROM "AWS/EC2" GROUP BY InstanceId`
+const memExpression = `SELECT AVG(mem_used_percent) FROM SCHEMA(CWAgent, InstanceId) GROUP BY InstanceId`
+
+// GetEC2Metrics gets the resource consumptions for EC2 instances
 func (c *Client) GetEC2Metrics(ctx context.Context, region string, interval time.Duration) error {
 	end := time.Now().UTC()
 	start := end.Add(-interval)
 
-	// Get the cpu consumption for all the instances in the region
-	err := c.getEC2CPU(ctx, region, start, end, interval)
+	period := int32(interval.Seconds())
+	// validate the casting from float64 to int32
+	if float64(period) != interval.Seconds() {
+		return fmt.Errorf("error casting %+v to int32", interval.Seconds())
+	}
```

Comment:
Is this necessary? Is there a specific case where casting to a float64 wil fail?

Reply:
I believe I added this because something came up where it didn't cast how it was expected... but that was a while ago and I don't remember now. Should have added a comment probably. But also if someone decides to use an interval of 2,147,483,648 seconds (or 68.1 years) it will overflow :joy: